### DISCUSSION TITLE
Lsp TCP support

### DISCUSSION THREAD

**Discussion by WannaBeKratos** [helix-editor/helix#4321] [Q&A] [answered]
Does helix support LSP using TCP connections or only stdin/out ?
There is no mention of socket support in the documentation.

**Comment by the-mikedavis** [comment, marked as answer]
No, but you can create a script that uses netcat to talk to the language server over TCP. See the top of https://github.com/helix-editor/helix/wiki/How-to-install-the-default-language-servers

**Reply by askazakov** [reply]
Now link leads to nowhere. Could you please actualise link? Or maybe now helix supports LSP via tcp ?

**Reply by askazakov** [reply]
I suppose now it is at https://github.com/helix-editor/helix/wiki/Language-Server-Configurations#introduction

> If your language server does not support stdio, you can use netcat as a drop-in proxy, just add this to your languages.toml:

```
[language-server.example-language-server]
command = "nc" 
args = ["127.0.0.1", "6008"]

[[language]]
name = "example-language"
language-servers = [ "example-language-server" ]
```
